### PR TITLE
feat: use stable trivy as a binary

### PR DESCRIPTION
**Pull request by pmengelbert**
**What this PR does / why we need it**:
Uses a trivy binary rather than attempting to build from their API. Trivy explicitly makes no guarantees about the stability of their internal API. It's a constantly moving target, and keeping our trivy up to date eats up a significant amount of time.

This is a draft PR to see how it does against the tests.

### CODE REVIEW

**Comment by pmengelbert** [comment]
- [ ] Supply config as cli args to trivy
- [ ] Disable all runtimes except containerd
- [ ] Fix unit tests

**Comment by pmengelbert** [comment]
- [ ] We need this https://github.com/aquasecurity/trivy/issues/4044

File: Dockerfile [hunk 1]
```diff
@@ -49,6 +50,39 @@ RUN \
...
+WORKDIR /build
+ENV GOCACHE=/root/gocache
+ENV CGO_ENABLED=0
+RUN \
+    --mount=type=cache,target=${GOCACHE} \
+    --mount=type=cache,target=/go/pkg/mod \
+<<EOF
+set -e
+git init
+git remote add origin ${TRIVY_REPO} || git remote set-url origin ${TRIVY_REPO}
```
Comment (sozercan) [inline Dockerfile:78]: we can validate and copy the binary instead of rebuilding

**Comment by pmengelbert** [comment]
I'll want to start from scratch once the Trivy changes are made

**Pull request closed without merging**